src/IECoreHoudini/SOP_SceneCacheSource.cpp

```diff
@@ -321,85 +329,123 @@ void SOP_SceneCacheSource::loadObjects( const IECoreScene::SceneInterface *scene
 		currentPath += scene->name().string();
 	}
 
-	if ( scene->hasObject() && UT_String( currentPath ).multiMatch( params.shapeFilter ) && tagged( scene, params.tagFilter ) )
+	if ( UT_String( currentPath ).multiMatch( params.shapeFilter ) && tagged( scene, params.tagFilter ) )
 	{
-		std::string name = relativePath( scene, rootSize );
-
-		Imath::M44d currentTransform;
-		if ( space == Local )
+		// filter out visibility or inherited visibility
+		if ( evalInt( pVisibilityFilter.getToken(), 0, 0 ) || inheritedVisibility )
```

Comment:
Why `|| inheritedVisibility`? Doesn't that mean we enter this block and do a bunch of unnecessary work when `pVisibilityFilter` is off?

Reply:
yes indeed